app/views/eventDetails.html

```diff
@@ -25,6 +25,7 @@
           type="submit"
           class="btn btn-success btn-lg"
           ng-click="saveEvent()"
+          ng-disabled="eventDetails.$invalid"
```

Comment:
You should be fine with using `eventDetails.$invalid` as there isn't any async functions for saving the data, but it might be best to use `!eventDetails.$valid` to keep it simular with the current code.

In case you didn;t know, here's an article which explains the difference between invalid and valid - https://itnext.io/valid-and-invalid-in-angular-forms-61cfa3f2a0cd